lib/tasks/puppet.rake

```diff
@@ -18,7 +18,7 @@ namespace :puppet do
         facts_stripped_of_class_names = YAML.load(puppet_facts.gsub(/\!ruby\/object.*$/, ''))
         User.as_anonymous_admin do
           host = Host::Managed.import_host(facts_stripped_of_class_names['name'], 'puppet')
-          host.import_facts(facts_stripped_of_class_names['values'].with_indifferent_access)
+          HostFactImporter.new(host).import_facts(facts_stripped_of_class_names['values'])
```

Comment:
A spotted a behavioural change: `with_indifferent_access` can you explain?

Reply:
One of the first lines of the method `import_facts` is `facts = facts.with_indifferent_access`.
No need to do so twice